torch/_higher_order_ops/cond.py

```diff
@@ -28,16 +30,130 @@
 )
 
 
+@contextmanager
+def _set_compilation_env():
+    _old_is_tracing = torch.fx._symbolic_trace._is_fx_tracing_flag
+    try:
+        # We need to turn off the is_fx_tracing_flag. Remove this flag check from dyanmo
+        # once we are confident fx tracing works with dynamo.
+        torch.fx._symbolic_trace._is_fx_tracing_flag = False
+        yield
+    finally:
+        torch.fx._symbolic_trace._is_fx_tracing_flag = _old_is_tracing
+
+
 @dataclass
 class UnsupportedAliasMutationException(RuntimeError):
     reason: str
 
 
+def cond(pred, true_branch, false_branch, operands):
+    r"""
+    `cond` can logically be seen as implemented as follows::
+
+    def cond(pred, true_branch, false_branch, operands):
+        if pred:
+            return true_branch(*operands)
+        else:
+            return false_branch(*operands)
+
+    Like a normal if statement in Python, users can use it to selectively execute a branch based on
+    the truth value of predicate in eager-mode or torch.compile. What make 'cond' unique is that
+    'cond' is exportable, lowerable: users can use it to capture a data-dependent control flow precisely,
+    preserve it accross the Pytorch stack and compile it in an environemnt without Python.
```

Comment:
Please change the intro to state the following. Let's clearly state that this feature is a prototype and may not work well outside of export:
```
Conditionally applies ``true_fn`` or ``false_fn``.

``cond`` is structured control flow operator. That is, it is like a Python if-statement, but has limitations on ``true_fn``, ``false_fn``, and ``operands`` that enable it to be capturable using torch.compile and torch.export.

Assuming the constraints on ``cond``'s arguments are met, ``cond`` is equivalent to the following::

    def cond(pred, true_branch, false_branch, operands):
        if pred:
            return true_branch(*operands)
        else:
            return false_branch(*operands)

.. warning::
   cond is a prototype feature in PyTorch, included as a part of the torch.export release. The main limitations are that it may not work in eager-mode PyTorch and you may encounter various failure modes while using it. Please look forward to a more stable implementation in a future version of PyTorch.

Read more about feature classification at: https://pytorch.org/blog/pytorch-feature-classification-changes/#prototype